app/ide-desktop/lib/client/src/bin/project-manager.ts

```
@@ -0,0 +1,60 @@
+import child_process, { SpawnOptions } from 'child_process'
+import * as config from 'config'
+import fss from 'node:fs'
+import { logger } from '../../../content/src/config'
+import util from 'node:util'
+const execFile = util.promisify(child_process.execFile)
+
+// =======================
+// === Project Manager ===
+// =======================
+
+/** Return the Project Manager path if it is valid. Otherwise, report an error and close the
+ * application. */
+export function pathOrPanic(args: config.Args): string {
```

It might be a matter of taste but I don't really like sthOrPanic style of functions, I'd rather throw an exception and allow caller to decide how it should be handled. (And also allow to provide more context information to be logged along the way.)

Also, it is not clear to me why "file not existing" is a panic, while other spawning issue "like file missing appropriate permission" is an exception?